src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -186,19 +231,27 @@ protected T WithOptions(ApplicationOptions applicationOptions)
         }
 
         /// <summary>
+        /// Sets the identifier of the software component (libraries/SDK) consuming MSAL.NET.
+        /// This will allow for disambiguation between MSAL usage by the app vs MSAL usage 
+        /// by component libraries. You can, for instance set it to the name of your application.
+        /// This is used in telemetry.
         /// </summary>
-        /// <param name="component"></param>
-        /// <returns></returns>
+        /// <param name="component">identifier of the software component (libraries/SDK) consuming MSAL.NET</param>
+        /// <returns>The builder to chain the .With methods</returns>
         public T WithComponent(string component)
         {
             Config.Component = GetValueIfNotEmpty(Config.Component, component);
             return (T)this;
         }
 
         /// <summary>
+        /// Sets a custom query parameters named slice= that may be sent to the STS for dogfood testing 
+        /// or debugging. 
+        /// Unless requested otherwise by Microsoft support, you should not set this parameter as it may have adverse effect on the application.
+        /// This property is also concatenated to the <c>extraQueryParameter</c> parameters of token acquisition operations.
         /// </summary>
-        /// <param name="sliceParameters"></param>
-        /// <returns></returns>
+        /// <param name="sliceParameters">name of the slice on which the application will run</param>
+        /// <returns>The builder to chain the .With methods</returns>
         public T WithSliceParameters(string sliceParameters)
```

Comment:
Do we want sliceParameters to be a Dictionary<string, string> like we do for ExtraQueryParameters?

Reply:
idk...i think sliceParameters is usually just one value, if they are setting more parameters, they should use ExtraQueryParameters. 

---
In reply to: [248388968](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/790#discussion_r248388968) [](ancestors = 248388968)